### PR TITLE
use python to process the configs

### PR DESCRIPTION
**Pull request by asakapab0i**

### CODE REVIEW

File: docker/zookeeper.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+            return server[1].split(':')[0]
+
+
+# Update the R53 DNS of the specific hosted zone
+def update_dns(domain, ip):
+    ZOO_HOSTED_ZONE = os.environ.get('ZOO_HOSTED_ZONE', False)
+    if not ZOO_HOSTED_ZONE:
+        exit('ZOO_HOSTED_ZONE is not defined')
+    logger.info("Updating the dns in route53.")
+    r53 = boto3.client('route53')
```
Comment (aren) [inline docker/zookeeper.py:112]: what are the permissions this role (`foxbox-zookeeper` or whatever it is) has for modifying route53 in AWS's IAM?

**Review by aren — COMMENTED** [review]
this looks really nice

File: docker/zookeeper.py (hunk 1 above)
Comment (asakapab0i) [reply docker/zookeeper.py:112]: Hi @aren,

This is only the permission required for us to update the DNS.

            "Effect": "Allow",
            "Action": "route53:ChangeResourceRecordSets"
            "Resource": "arn:aws:route53:::hostedzone/HOSTED_ZONE_ID"

And each environment has a specific resource set and we have to create a policy for each.

File: docker/Dockerfile [hunk 2]
```diff
@@ -11,7 +11,8 @@ ENV ZOO_CONF_DIR=/conf \
     ZOO_AUTOPURGE_SNAPRETAINCOUNT=3 \
     ZOO_MAX_CLIENT_CNXNS=60 \
     ZOO_STANDALONE_ENABLED=true \
-    ZOO_ADMINSERVER_ENABLED=true
+    ZOO_ADMINSERVER_ENABLED=true \
+    JVMFLAGS=-Dsun.net.inetaddr.ttl=0
```
Comment (grk-it) [inline docker/Dockerfile:15]: curious, why did we add this flag ?

File: docker/zookeeper.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+# Get the environment variables
+def get_config_env():
+    ZOO_DATA_DIR = os.environ.get('ZOO_DATA_DIR')
+    ZOO_DATA_LOG_DIR = os.environ.get('ZOO_CONF_DIR')
+    ZOO_TICK_TIME = os.environ.get('ZOO_TICK_TIME', 2000)
+    ZOO_INIT_LIMIT = os.environ.get('ZOO_INIT_LIMIT', 5)
+    ZOO_SYNC_LIMIT = os.environ.get('ZOO_SYNC_LIMIT', 2)
+    ZOO_AUTOPURGE_SNAPRETAINCOUNT = os.environ.get('ZOO_AUTOPURGE_SNAPRETAINCOUNT', 3)
+    ZOO_AUTOPURGE_PURGEINTERVAL = os.environ.get('ZOO_AUTOPURGE_PURGEINTERVAL', 0)
+    ZOO_MAX_CLIENT_CNXNS = os.environ.get('ZOO_MAX_CLIENT_CNXNS', 60)
```
Comment (grk-it) [inline docker/zookeeper.py:19]: I think we need to increase this number. In the context of prod, especially during deployment the number of nodes can go upto 60 ?

File: docker/zookeeper.py [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+    config.touch(exist_ok=True)
+    # Write to the file
+    with open('{}/myid'.format(ZOO_DATA_DIR), 'w') as conf:
+        logger.info("Writing the contents of the config file.")
+        conf.write(myid)
+
+
+# Get myid from using the current subnet
+def get_myid():
+    private_ip = requests.get(os.environ['ECS_CONTAINER_METADATA_URI_V4'], timeout=0.01).json()['Networks'][0]['IPv4Addresses'][0]
```
Comment (grk-it) [inline docker/zookeeper.py:92]: question: when the containers come up, are they always assigned nodes with a different subnet. How is this ensured.

File: docker/Dockerfile [hunk 5]
```diff
@@ -80,6 +85,8 @@ EXPOSE 2181 2888 3888 8080
 ENV PATH=$PATH:/$DISTRO_NAME/bin \
     ZOOCFGDIR=$ZOO_CONF_DIR
 
-COPY docker-entrypoint.sh /
-ENTRYPOINT ["/docker-entrypoint.sh"]
+COPY zookeeper.py .
+COPY docker-entrypoint.sh .
+
+ENTRYPOINT [ "./docker-entrypoint.sh" ]
 CMD ["zkServer.sh", "start-foreground"]
```
Comment (grk-it) [inline docker/Dockerfile:92]: I think, this is what we need for the `helix-ui-server` as well right.

File: docker/Dockerfile (hunk 2 above)
Comment (asakapab0i) [reply docker/Dockerfile:15]: It's for the DNS resolution inside the container, sometimes the old IP address sticks to the process and it hangs the system.

Aren suggested it.

File: docker/zookeeper.py (hunk 3 above)
Comment (asakapab0i) [reply docker/zookeeper.py:19]: These can be configured in the task definition.

File: docker/zookeeper.py (hunk 4 above)
Comment (asakapab0i) [reply docker/zookeeper.py:92]: Yes it get's assigned evenly and so the number of container we have depends on upon the # of subnets in the VPC.

File: docker/Dockerfile (hunk 5 above)
Comment (asakapab0i) [reply docker/Dockerfile:92]: Most likely, I have to look at it again.

File: docker/zookeeper.py (hunk 4 above)
Comment (grk-it) [reply docker/zookeeper.py:92]: Thanks Bryan for the detailed discussion offline!

File: docker/zookeeper.py (hunk 1 above)
Comment (aren) [reply docker/zookeeper.py:112]: Ok great. I just want to make sure this role can only update this service's DNS entries and it appears that's how you have configured it. Correct?

Comment (asakapab0i) [reply docker/zookeeper.py:112]: Yup, that is correct, it won't update any DNS other than specified.

**Review by aren — APPROVED** [review]

**Pull request merged**